### PR TITLE
Reorganization of the GenFilters package

### PR DESCRIPTION
**Pull request by fabiocos**
#### PR description:

Move all plugins into a ```plugins```, adjust the BuildFile accordingly (with a minimal cleaning), as discussed with @smuzaffar and in the review of #28201 (which should be adapted to this PR at this point).

#### PR validation:

Code compiles, test wf 541 provides same event output.

### CODE REVIEW

**Comment by fabiocos** [comment]
please test workflow 541.0

**Comment by fabiocos** [comment]
please test workflow 541.0

File: GeneratorInterface/GenFilters/BuildFile.xml [hunk 1]
```diff
@@ -1,30 +1,3 @@
...
-<use   name="CommonTools/BaseParticlePropagator"/>
-<use   name="TrackingTools/GeomPropagators"/>
-<use   name="DataFormats/HepMCCandidate"/>
-<use   name="DataFormats/JetReco"/>
-<use   name="DataFormats/EgammaReco"/>
-<use   name="DataFormats/Math"/>
-<flags   EDM_PLUGIN="1"/>
+<export>
+  <lib name="1"/>
+</export>
```
Comment (smuzaffar) [inline GeneratorInterface/GenFilters/BuildFile.xml]: @fabiocos , there is no need to have this BuildFile. BuildFile in package directory is only needed when there are source files in package/{src,interface} directory.

Comment (fabiocos) [reply GeneratorInterface/GenFilters/BuildFile.xml]: @smuzaffar ok, thanks for the remark

**Comment by fabiocos** [comment]
please test with cms-sw/cmssw#28368

to also verify the removal of reproducibility issues, this PR is totally independent

**Comment by qliphy** [comment]
+1

**Comment by fabiocos** [comment]
please test

**Comment by fabiocos** [comment]
+1

the test is just to verify the effect of integrating #28368 in the baseline

**Comment by fabiocos** [comment]
merge

**Comment by Dr15Jones** [comment]
This has triggered the duplicate plugin registration warnings in the IB.

**Comment by fabiocos** [comment]
@Dr15Jones yes, this is why I asked @smuzaffar to trigger a full rebuild, that has cleaned the situation

**Pull request merged**